fix: arrumando dependências do ubuntu

yanrdgs-dev opened the pull request:
Troca das dependências do python3-dbus. No apt, essa package não existe, então troquei para instalar cada uma das dependências do dbus separadamente. Além disso, fiz uma função para checagem do keyring do google chrome, já que o apt não consegue encontrar o google-chrome-stable sem esse keyring.

yanrdgs-dev committed:
fix: arrumando dependências do ubuntu

yanrdgs-dev commented:
Além disso, talvez seja necessário fazer uma lógica para rodar um venv para o programa. Com o Ubuntu, recebi o erro de que os pacotes do sistemas são gerenciados externamente. Para instalar as dependências, precisei criar um venv para rodar o pip install. Se julgar necessário, posso fazer essa lógica da instalação também.

marcusmartinss commented on install.sh:
```diff
@@ -12,15 +12,37 @@ distro=$(grep "^ID=" /etc/os-release | awk -F= '{print tolower($2)}' | tr -d '"'
+# função para checar de keyring do Google Chrome existe
+# ao rodar o install.sh, tive erros pra instalar o google-chrome-stable. 
+# o erro era que o pacote não existia, então adicionei essa função para instalar o keyring caso não seja encontrado
+check_google_keyring() {
```
testei aqui no meu e não funcionou no meu Fedora, pensando aqui que dava pra fazer só um `whereis google-chrome` e verficar a saída que ele vai saber onde está o binário do Chrome.

yanrdgs-dev replied:
o dnf possui o google-chrome-stable? o problema do Ubuntu foi que o google-chrome-stable não existia no repositório do Ubuntu, e precisaria importar o repositório do Chrome.

Acho que o whereis não seria tão necessário na verdade, visto que mesmo se já tiver um outro google-chrome-stable instalado, o apt vai detectar.

marcusmartinss commented on install.sh:
```diff
@@ -12,15 +12,37 @@ distro=$(grep "^ID=" /etc/os-release | awk -F= '{print tolower($2)}' | tr -d '"'
 # Define os pacotes com base nas distros-mãe
 pacotes_debian_base="pulseaudio-utils dbus-python-devel python3-dbus google-chrome-stable"
+# python3-dbus não existe no apt. então, é necessário instalar as bibliotecas do dbus e suas dependências separadamente
+pacotes_ubuntu_base="pulseaudio-utils build-essential libdbus-glib-1-dev libgirepository1.0-dev google-chrome-stable"
```
Boa noite Yan, achei o pacote [aqui na documentação](https://packages.debian.org/search?keywords=python3-dbus#:~:text=Exact%20hits-,Package%20python3%2Ddbus,-buster%20(oldoldstable)) do repositório do Debian, mas vou fazer uns testes aqui no Ubuntu antes, geralmente o que funciona no Debian funciona no Ubuntu

marcusmartinss replied:
Qual foi o erro que apareceu para você? Porque consegui encontrar o pacote e rodar o install.sh no Ubuntu aqui

![image](https://github.com/user-attachments/assets/e338a079-9da4-40ae-a2f2-3b7a6becb49a)

yanrdgs-dev replied:
Realmente Marcus, acho que só me confundi aqui. Porém, o dbus-python-devel realmente não existe, mas não sei a influência dele no arquivo principal. Perdão pelo erro, apaguei a variável de pacotes_ubuntu_base.

marcusmartinss commented:
> Além disso, talvez seja necessário fazer uma lógica para rodar um venv para o programa. Com o Ubuntu, recebi o erro de que os pacotes do sistemas são gerenciados externamente. Para instalar as dependências, precisei criar um venv para rodar o pip install. Se julgar necessário, posso fazer essa lógica da instalação também.

Se você acha uma boa ideia, pode adicionar! Quanto mais estável melhor

marcusmartinss reviewed (changes requested):
Deixei alguns comentários para saber mais sobre os problemas que estava tendo, podemos ir consertando o problema juntos

Resolver:
- [x] Dificuldade de encontrar `python3-dbus`
- [x] Simplificar verificação do chrome na máquina
- [x] Confirmar se é necessário mesmo separar instalação Ubuntu do Debian

yanrdgs-dev committed:
fix: arrumando lógica da instalação do chrome, juntando debian e ubuntu novamente

yanrdgs-dev committed:
fix: colocando de volta dbus-python-devel

Não sei a necessidade do dbus-python-devel no arquivo principal, optei por deixar.

yanrdgs-dev commented:
Fiz as alterações agora. Coloquei um whereis para verificar se o google-chrome-stable existe, para evitar que se importe o repositório da Google pra nada.

Também, fiz alterações no `wget` e no `sudo tee` (`-qO` e `/dev/null` para saídas silenciosas nos respectivos comandos).

Não quis alterar a questão da lógica do venv agora, pois fiquei pensando depois que, caso seja utilizada essa alternativa pro pip install, o usuário precisaria dar `source venv/bin/activate` toda vez que fosse iniciar o programa. Aqui vejo 2 soluções:
-  `pip install` sendo com a tag `--user`, para instalação local, assim o python não retorna erro, mas precisaria adicionar o `~/.local/bin` ao PATH do sistema.
- Criar um outro shell script para inicializar o programa. Nele, o ambiente virtual seria ativado antes de iniciar o programa. Algo como: 
```bash
#!/bin/bash

# vaga-extraordinaria.sh
source venv/bin/activate
python3 main.py
```

Confirmei que não é necessário a separação de Debian pro Ubuntu, o python3-dbus realmente existe e não dá problemas. Porém, o dbus-python-devel não existe, mas não dá erro na execução do script, apenas não é instalado.
